src/main/java/org/candlepin/subscriptions/db/model/Host.java

```diff
@@ -177,8 +177,10 @@ public void populateFieldsFromHbi(InventoryHostFacts inventoryHostFacts,
         this.subscriptionManagerId = inventoryHostFacts.getSubscriptionManagerId();
         this.guest = normalizedFacts.isVirtual();
         this.hypervisorUuid = normalizedFacts.getHypervisorUuid();
-        this.measurements.put(Measurement.Uom.CORES, normalizedFacts.getCores().doubleValue());
-        this.measurements.put(Measurement.Uom.SOCKETS, normalizedFacts.getSockets().doubleValue());
+        this.measurements.put(Measurement.Uom.CORES,
+            (normalizedFacts.getCores() == null ? 0.0 : normalizedFacts.getCores().doubleValue()));
+        this.measurements.put(Measurement.Uom.SOCKETS,
+            (normalizedFacts.getSockets() == null ? 0.0 : normalizedFacts.getSockets().doubleValue()));
```

Comment:
I would rather see us check these values and *not* call `this.measurements.put` if the value is `null`. Simple `if` checks would suffice.

I would also like to see a unit test for this, so that we don't regress on either cores or sockets.

Reply:
Very good, I'll change this and add a unit test.